themes/hugo-book-custom/layouts/partials/docs/menu-filetree.html

```diff
@@ -4,25 +4,52 @@
 {{ end }}
 
 {{ with .Site.GetPage $bookSection }}
-  {{ template "book-section-children" (dict "Section" . "CurrentPage" $) }}
+    <sl-tree>
+    {{ template "book-section-children" (dict "Section" . "CurrentPage" $) }}
+    </sl-tree>
+
+    <script>
+    document.addEventListener("DOMContentLoaded", function () {
+        const activeLink = document.querySelector("sl-tree-item a.active");
+
+        var getParents = function (elem, selector) {
+	        var parents = [];
+	        for ( ; elem && elem !== document; elem = elem.parentNode ) {
+		        if (selector) {
+			        if (elem.matches(selector)) {
+				        parents.push(elem);
+			        }
+			        continue;
+		        }
+		        parents.push(elem);
+	        }
+	        return parents;
```

Comment:
you seem to be supporting two cases here; with a selector, and without. Just make the function only support a selector, seems easier.

Reply:
modified